automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/library/DataLibrarySpec.scala

```diff
@@ -51,4 +51,58 @@ class DataLibrarySpec extends FreeSpec with WebBrowserSpec with UserFixtures wit
       }
     }
   }
+
+  /**
+    * This test creates two workspace associated with different tags. Search for tag should
+    *  only find right workspace in Data Library table.
+    */
+  "Use a known tag to search for dataset" in withWebDriver { implicit driver =>
+    val curatorUser = UserPool.chooseCurator
+    implicit val authToken: AuthToken = curatorUser.makeAuthToken()
+    withWorkspace(namespace, "DataLibrarySpec_tags_", attributes = Some(WorkspaceData.tagA)) { wsName1 =>
+      withWorkspace(namespace, "DataLibrarySpec_tags_", attributes = Some(WorkspaceData.tagB)) { wsName2 =>
+        withCleanUp {
+
+          val tag1: String = WorkspaceData.tagA.get("tag:tags").get(0)
+          val tag2: String = WorkspaceData.tagB.get("tag:tags").get(0)
+          val data1 = LibraryData.metadataBasic + ("library:datasetName" -> wsName1)
+          api.library.setLibraryAttributes(namespace, wsName1, data1)
+          val data2 = LibraryData.metadataBasic + ("library:datasetName" -> wsName2)
+          api.library.setLibraryAttributes(namespace, wsName2, data2)
+          register cleanUp {
```

Comment:
We have enough tests that publish/unpublish. It's probably time to introduce a `withPublishedWorkspace` or build it into `withWorkspace`. That would remove the need to use `withCleanUp` and manually registering the unpublish clean-up.